### DISCUSSION TITLE
Fake S3 bucket for local-dev

### DISCUSSION THREAD

**Discussion by shuntl** [wasp-lang/wasp#1988] [Archived]
It would be awesome to have wasp in development mode be able to spin up a dummy S3 bucket. Then prod env we can add the details for either a real S3, Digital Ocean space etc etc.

**Comment by Martinsos** [comment]
Huh that would be interesting indeed!

It is a bit specific though hm, doesn't sound like a feature to add to the core language of Wasp.

But maybe concept of FileStorage would be interesting, and then you can choose S3, Azure Blob, ... .
Ok yeah that is actually what you said, I get it now! Ok, that would be quite interesting, you are right.

I wonder what the right name would be. FileStorage? Storage?

We could have a way to say we want to have Storage, and it can be Dummy for start, and then as you said it can be S3 or sometihng else. We would need to have abstraction layer over it though. I wonder if a library for this already exists. And they would still need to be able to access it under the hood when needed (and use e.g. specific S3 options).
